cli/src/main/java/org/openjdk/skara/cli/GitWebrev.java

```diff
@@ -251,10 +265,9 @@ private static void generate(String[] args) throws IOException {
                         var head = repo.head();
                         var shortestDistance = -1;
                         var pullPath = repo.pullPath(remote);
-                        var remoteBranches = repo.remoteBranches(remote);
-                        for (var remoteBranch : remoteBranches) {
-                            var fetchHead = repo.fetch(URI.create(pullPath), remoteBranch.name());
-                            var mergeBase = repo.mergeBase(fetchHead, head);
+                        for (var branch : repo.branches(remote)) {
+                            var branchHead = repo.resolve(branch).orElseThrow();
+                            var mergeBase = repo.mergeBase(branchHead, head);
```

Comment:
Now that the fetch is gone, could repo become a ReadOnlyRepository again?

Reply:
Ah right, yes! Will fix :+1: